spec/prog/postgres/converge_postgres_resource_spec.rb

```diff
@@ -3,403 +3,484 @@
+    pr = PostgresResource.create(
+      name: "pg-test", superuser_password: "dummy-password", ha_type: "sync",
+      target_version: "17", location_id: location_id, project_id: project.id,
+      user_config: {}, pgbouncer_user_config: {}, target_vm_size: "standard-2",
+      target_storage_size_gib: 64, private_subnet_id: private_subnet.id
+    )
+    Strand.create_with_id(pr, prog: "Postgres::PostgresResourceNexus", label: "wait")
+    pr
+  }
+
+  let(:strand) {
+    Strand.create(
+      prog: "Postgres::ConvergePostgresResource", label: "start",
+      parent_id: postgres_resource.strand.id,
+      stack: [{"subject_id" => postgres_resource.id}]
     )
   }
 
   before do
     allow(nx).to receive(:postgres_resource).and_return(postgres_resource)
   end
 
+  def create_server(version: "17", representative: false, vm_host_data_center: nil, timeline: self.timeline, timeline_access: "fetch", resource: postgres_resource)
+    vm_host = create_vm_host(location_id: resource.location_id, data_center: vm_host_data_center)
+    vm = Prog::Vm::Nexus.assemble_with_sshable(
+      project.id, name: "pg-vm-#{SecureRandom.hex(4)}", private_subnet_id: resource.private_subnet_id,
+      location_id: resource.location_id, unix_user: "ubi"
+    ).subject
+    vm.update(vm_host_id: vm_host.id)
+    # Add storage volume to match target_storage_size_gib so needs_recycling? returns false by default
+    VmStorageVolume.create(vm_id: vm.id, size_gib: resource.target_storage_size_gib, boot: false, disk_index: 1)
+    server = PostgresServer.create(
+      timeline: timeline, resource_id: resource.id, vm_id: vm.id,
+      representative_at: representative ? Time.now : nil,
+      synchronization_status: "ready", timeline_access: timeline_access, version: version
+    )
+    Strand.create_with_id(server, prog: "Postgres::PostgresServerNexus", label: "wait")
+    server
+  end
+
   it "exits if destroy is set" do
     expect(nx).to receive(:when_destroy_set?).and_yield
     expect { nx.before_run }.to exit({"msg" => "exiting early due to destroy semaphore"})
   end
 
   describe "#start" do
     it "naps if read replica parent is not ready" do
-      parent = instance_double(PostgresResource, ready_for_read_replica?: false)
-      allow(postgres_resource).to receive_messages(read_replica?: true, parent: parent)
+      parent = PostgresResource.create(
+        name: "pg-parent", superuser_password: "dummy-password", ha_type: "none",
+        target_version: "17", location_id: location_id, project_id: project.id,
+        user_config: {}, pgbouncer_user_config: {}, target_vm_size: "standard-2",
+        target_storage_size_gib: 64, private_subnet_id: private_subnet.id
+      )
+      Strand.create_with_id(parent, prog: "Postgres::PostgresResourceNexus", label: "wait")
+      postgres_resource.update(parent_id: parent.id)
+      allow(parent).to receive(:ready_for_read_replica?).and_return(false)
+      allow(postgres_resource).to receive(:parent).and_return(parent)
```

Comment:
I think this is unnecessary with the added `postgres_resource.update(parent_id: parent.id)` above.

```suggestion
```

Similar issue in following spec.